app/src/main/java/com/example/henzoshimada/feeltrip/Mood.java

```diff
@@ -219,9 +263,11 @@ public Date getDate() {
     }
 
     public void setDate(Date date) {
-        this.date = date.getTime();
-        this.trueDate = date;
-        setStateByIndex(2);
+        if (this.date != date.getTime()) {
+            this.date = date.getTime();
+            this.trueDate = date;
+            setStateByIndex(2);
```

Comment:
To ensure we don't set the Date field as being "edited" when really we just passed the exact same Date into the Mood. This check is only relevant for the purposes of editing a mood post. For creating a new post, it will always be overwritten by the current system time.